craft_parts/lifecycle_manager.py

```diff
@@ -103,6 +103,40 @@ def project_info(self) -> ProjectInfo:
         """Obtain information about this project."""
         return self._project_info
 
+    def clean(
+        self, step: Optional[Step] = None, *, part_names: List[str] = None
+    ) -> None:
+        """Clean the specified step and parts.
+
+        Cleaning a step removes its state and all artifacts generated in that
+        step and subsequent steps for the specified parts.
+
+        :para step: The step to clean.
+        :param part_names: The list of part names to clean. If not specified,
+            all parts will be cleaned and work directories will be removed.
+        """
+        if not step:
+            step = Step.PULL
+
+        self._executor.clean(initial_step=step, part_names=part_names)
+
+    def refresh_packages_list(self, *, system=False) -> None:
+        """Update the available packages list.
+
+        The list of available packages should be updated before planning the
+        sequence of actions to take. To ensure consistency between the scenarios,
+        it shouldn't be updated between planning and execution.
+
+        :param system: Also refresh the list of available build packages to
+            install on the host system.
+        """
+        packages.Repository.refresh_stage_packages_list(
+            application_name=self._application_name, target_arch=self._target_arch
+        )
+
+        if system:
```

Comment:
Is it expected to be called more than once with a conditional?  Maybe they should just be separate?

Reply:
We'll also need to update the overlay package list and my idea was to consolidate all three in a single call to simplify the API. Let's see how the overlay discussion goes, we can refactor later if needed.